src/cplscheme/CompositionalCouplingScheme.cpp

```diff
@@ -25,18 +25,61 @@ T min(T a, T b)
 
 namespace precice::cplscheme {
 
+namespace {
+bool compatibleTimeWindowSizes(const CouplingScheme &impl, const CouplingScheme &expl)
+{
+  if (!impl.hasTimeWindowSize() || !expl.hasTimeWindowSize()) {
+    return true;
+  }
+  double idt = impl.getTimeWindowSize();
+  double edt = expl.getTimeWindowSize();
+  if (math::equals(edt, idt)) {
+    return true; // equal is compatible
+  }
+  if (math::smaller(edt, idt)) {
+    return false; // smaller edt is incompatible
+  }
+  return math::equals(std::remainder(edt, idt), 0.0);
```

Comment:
Just as a side comment: the many returns and nested if make the code rather hard to follow. The code here could, for example, be simplified as

```suggestion
return math::equals(explTime, implTime) || math::equals(std::remainder(explTime, implTime), 0.0);
```

Not sure which of these two versions is easier to read.

Reply:
This can definitely be shortened. The verbose way is easier to document while developing.